xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -129,15 +102,33 @@ public String toString() {
       } else {
         toStringHelper.add("virtualHosts", virtualHosts);
       }
-      if (hasFaultInjection) {
-        toStringHelper.add("faultInjectionEnabled", true)
-            .add("httpFault", httpFault);
+      if (filterChain != null) {
+        toStringHelper.add("filterChain", filterChain);
       }
       if (listener != null) {
         toStringHelper.add("listener", listener);
       }
       return toStringHelper.toString();
     }
+
+    static final class NamedFilter {
```

Comment:
Why this is defined here? Instead of in Filter? Also, this makes the definitions of Filter and FilterConfig very confusing. Why not just add the name property to `FilterConfig`?

Reply:
> Why this is defined here? Instead of in Filter? 

Because it's only used for constructing or accessing data of `LdsUpdate`.

> Why not just add the name property to FilterConfig?

The parsed config returned by the API `StructOrError<FilterConfig> parseFilterConfig(Message rawProtoMessage) ` is not tied to a filter name.  Actually it's possible that two filter instances with different names can have the same FilterConfig. It's just the pure config data parsed from envoy proto.

Reply:
> Actually it's possible that two filter instances with different names can have the same FilterConfig.

That's just two separate FilterConfig instances, I don't see the difference between putting the name property on FilterConfig vs wrapping with another layer. Anyway, I accept this approach, but `NamedFilter` shouldn't be defined here. 

> Because it's only used for constructing or accessing data of LdsUpdate.

That's not a valid reason, `NamedFilter` is useless without the feature of HttpFilter, it's HttpFilter-specific. Putting it here is anti-modularity (Filter and other Filter-related classes are defined elsewhere while this one is defined on XdsClient surface). I don't think this is just a style issue...

Reply:
Done.